### PR TITLE
Add option to point to a custom plugin configurator

### PR DESCRIPTION
**Pull request by pmeijer**
The `metadata.json` associated with a plugin now takes an optional parameter, `configWidget`, with a path to an AMD (requirejs module). If defined the InterpeterManager in the generic UI will call that instead of the default config dialog. 

For API of configurator see `src/plugin/coreplugins/CustomPluginConfig/widget/ConfigWidget.js`.

Note that you can reuse the default dialog by including and calling it from your custom widget.

### CODE REVIEW

File: src/client/js/Utils/InterpreterManager.js [hunk 1]
```diff
@@ -45,77 +45,105 @@ define([
...
+    InterpreterManager.prototype.loadConfigDialog = function (metadata, callback) {
+        if (!metadata.configWidget) {
+            callback(null, PluginConfigDialog);
+        } else {
+            requirejs([metadata.configWidget],
+                function (CustomConfigDialog) {
+                    callback(null, CustomConfigDialog);
+                },
+                function (err) {
+                    callback(err);
```
Comment (kecso) [inline src/client/js/Utils/InterpreterManager.js]: Couldn't it fall back to the PluginConfigDialog?
Maybe with an extra notification...

File: src/plugin/coreplugins/CustomPluginConfig/widget/ConfigWidget.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*globals, define, $, WebGMEGlobal*/
```
Comment (kecso) [inline src/plugin/coreplugins/CustomPluginConfig/widget/ConfigWidget.js]: A base class would be nice.

File: src/client/js/Utils/InterpreterManager.js (hunk 1 above)
Comment (pmeijer) [inline src/client/js/Utils/InterpreterManager.js]: I think it's better fail hard here. The plugin might be unusable without the custom configuration.

File: src/plugin/coreplugins/CustomPluginConfig/widget/ConfigWidget.js (hunk 2 above)
Comment (pmeijer) [inline src/plugin/coreplugins/CustomPluginConfig/widget/ConfigWidget.js]: The API is so small and the base class wouldn't really provide any functionality.

**Pull request merged**